nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureBlobStorage.java

```diff
@@ -0,0 +1,193 @@
+        @WritesAttribute(attribute = "azure.secondaryUri", description = "Secondary location for blob content"), @WritesAttribute(attribute = "azure.etag", description = "Etag for the Azure blob"),
+        @WritesAttribute(attribute = "azure.length", description = "Length of the blob"), @WritesAttribute(attribute = "azure.timestamp", description = "The timestamp in Azure for the blob"),
+        @WritesAttribute(attribute = "mime.type", description = "MimeType of the content"), @WritesAttribute(attribute = "lang", description = "Language code for the content"),
+        @WritesAttribute(attribute = "azure.blobtype", description = "This is the type of blob and can be either page or block type") })
+@Stateful(scopes = { Scope.LOCAL, Scope.CLUSTER }, description = "After performing a listing of blobs, the timestamp of the newest blob is stored. "
+        + "This allows the Processor to list only blobs that have been added or modified after " + "this date the next time that the Processor is run.")
+public class ListAzureBlobStorage extends AbstractListProcessor<BlobInfo> {
+
+    private static final PropertyDescriptor PREFIX = new PropertyDescriptor.Builder().name("Prefix").description("Search prefix for listing").addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true).required(false).build();
+
+    public static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(AzureConstants.ACCOUNT_NAME, AzureConstants.ACCOUNT_KEY, AzureConstants.CONTAINER, PREFIX));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    protected Map<String, String> createAttributes(BlobInfo entity, ProcessContext context) {
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put("azure.etag", entity.getEtag());
+        attributes.put("azure.primaryUri", entity.getPrimaryUri());
+        attributes.put("azure.secondaryUri", entity.getSecondaryUri());
+        attributes.put("azure.blobname", entity.getName());
+        attributes.put("azure.blobtype", entity.getBlobType());
+        attributes.put("azure.length", String.valueOf(entity.getLength()));
+        attributes.put("azure.timestamp", String.valueOf(entity.getTimestamp()));
+        attributes.put("mime.type", entity.getContentType());
+        attributes.put("lang", entity.getContentLanguage());
+
+        return attributes;
+    }
+
+    @Override
+    protected String getPath(final ProcessContext context) {
+        return context.getProperty(AzureConstants.CONTAINER).evaluateAttributeExpressions().getValue();
+    }
+
+    @Override
+    protected boolean isListingResetNecessary(final PropertyDescriptor property) {
+        // TODO - implement
+        return false;
+    }
+
+    @Override
+    protected Scope getStateScope(final ProcessContext context) {
+        return Scope.CLUSTER;
```

Comment:
OK.  I can see how you'd want to force cluster state, since the files are being pulled from a remote location.  Makes it easier for the user to not have to worry about the possible confusion of local vs cluster state and primary-only scheduling.